scenarios/throughput_stress.go

```diff
@@ -0,0 +1,111 @@
+package scenarios
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"go.temporal.io/api/workflowservice/v1"
+	"sync/atomic"
+	"time"
+
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/operatorservice/v1"
+	"go.temporal.io/api/serviceerror"
+
+	"github.com/temporalio/omes/loadgen"
+	"github.com/temporalio/omes/loadgen/throughputstress"
+	"go.temporal.io/sdk/client"
+)
+
+// --option arguments
+const (
+	IterFlag     = "internal-iterations"
+	CANEventFlag = "continue-as-new-after-event-count"
+)
+
+const ThroughputStressScenarioIdSearchAttribute = "ThroughputStressScenarioId"
+
+type tpsExecutor struct {
+	workflowCount atomic.Uint64
+}
+
+func (t *tpsExecutor) Run(ctx context.Context, info loadgen.ScenarioInfo) error {
+	// Make sure the search attribute is registered
+	attribMap := map[string]enums.IndexedValueType{
+		ThroughputStressScenarioIdSearchAttribute: enums.INDEXED_VALUE_TYPE_KEYWORD,
+	}
+	_, err := info.Client.OperatorService().AddSearchAttributes(ctx,
+		&operatorservice.AddSearchAttributesRequest{
+			Namespace:        info.Namespace,
+			SearchAttributes: attribMap,
+		})
+	var svcErr *serviceerror.AlreadyExists
+	if !errors.As(err, &svcErr) {
+		return err
+	}
+
+	genericExec := &loadgen.GenericExecutor{
+		DefaultConfiguration: loadgen.RunConfiguration{
+			Iterations:    20,
+			MaxConcurrent: 5,
+		},
+		Execute: func(ctx context.Context, run *loadgen.Run) error {
+			internalIterations := run.ScenarioInfo.ScenarioOptionInt(IterFlag, 5)
+			continueAsNewCount := run.ScenarioInfo.ScenarioOptionInt(CANEventFlag, 100)
```

Comment:
I'm wondering why we need continue-as-new and child workflows to test throughput.
It's not _wrong_ but omes exposes metrics for workflow e2e latency and throughput that we won't be able to leverage with this approach.

Reply:
I don't disagree. I'm just recreating the scenario. Really, the name `throughputstress` is probably just not precise.